packages/braid-design-system/src/lib/components/Dialog/Dialog.tsx

```diff
@@ -13,22 +15,42 @@ const modalStyle = {
 } as const;
 
 export interface DialogProps
-  extends Omit<ModalProps, keyof typeof modalStyle | 'width'> {
+  extends Omit<ModalProps, keyof typeof modalStyle | 'id' | 'width'> {
+  id?: ModalProps['id'];
   width?: ModalProps['width'];
 }
 
-export const Dialog = ({ width = defaultWidth, ...restProps }: DialogProps) => (
-  <Modal width={width} {...restProps} {...modalStyle} />
-);
+export const Dialog = ({
+  id,
+  width = defaultWidth,
+  ...restProps
+}: DialogProps) => {
+  const fallbackId = useId();
+  const resolvedId = id || fallbackId;
```

Comment:
Did you consider handling this once down in `ModalContent` and making that `id` optional?

That would make `id` just a passthrough for all these higher level `Dialog`, `DialogContent`, `Drawer` and `DrawerContent` components.